language/move-model/src/spec_translator.rs

```diff
@@ -577,7 +598,7 @@ impl<'a, 'b, T: ExpGenerator<'a>> ExpRewriterFunctions for SpecTranslator<'a, 'b
 
     fn rewrite_temporary(&mut self, id: NodeId, idx: TempIndex) -> Option<Exp> {
         // Compute the effective index.
-        let is_mut = self.fun_env.get_local_type(idx).is_mutable_reference();
+        let is_mut = self.builder.get_local_type(idx).is_mutable_reference();
```

Comment:
This is a bug fix. As the `builder` may have more temporaries than what we recorded in `fun_env`.

Reply:
Well, this is probably not a bug, using `fun_env` and `builder` gives different types for the same local idx in a way I did not fully understand... The unit test failure is caused by this and I am still checking.

Reply:
Last update:
- use `fun_env` if `idx < param_count`: we want to be aware whether the param is a mut ref or not.
- use `builder` otherwise.